### ISSUE TITLE
Better API?

### ISSUE THREAD

**Issue by ivanjaros** [couchbase/moss#56]
In comparison with BoltDB or Badger, MOSS has quite unpleasing and confusing API. Any plans for improvement?

**Comment by mschoch** [comment]
It's important to remember that moss was not designed to be a key/value store originally.  It started as a caching layer in-front of key/value stores, and then eventually evolved into supporting it's own persistence.  And even at it's current stage it's really designed for asynchronous persistence, and not durable writes like most other key/value stores, so it's API tends to reflect that fact as well.  I have always thought that we could introduce a new veneer or porcelain layer on top of moss that cleaned up some of these details.

In order to be more constructive, it would be helpful to provide specific examples of operations that are simpler/easier/more-clearly expressed in other APIs so that we can give thought to addressing them.

**Comment by ivanjaros** [comment]
The naming is weird. I was looking for "open" method, but there is none. It seems that's represented with OpenStoreCollection / OpenCollection. I have no idea what is the difference between Store and Collection that these methods return. I found Set() method in "batch". So ok, I tried that and  .. nothing happened. No files created, no data persisted. So I was looking for "commit" or "save" maybe, which might be needed after the batch but I see nothing.

So at this point I am not able to use this library at all :D

Bolt has simple Open to initialize connection/instance, it has View/Update transactions for reading/writing. That is intuitive. Then there is Put() or Set(). Really simple and no guessing required. Additionally Bolt has buckets, which I presume is like collection in moss? It also has the same methods.
Badger is basically the same.

**Comment by mschoch** [comment]
Right, the problems is actually doesn't even support the use case.  It has no commit method, there is no way to guarantee your data makes it to disk other than to wait.  So, the bigger problem is you're trying to use it for something it wasn't intended for.
